### DISCUSSION TITLE
ARM: Mount to path

### DISCUSSION THREAD

**Discussion by overheadhunter** [winfsp/winfsp#447] [Q&A]
Another day, another question:

We encountered some problems when trying to mount to a path (as opposed to drive letter). Using memfs shows this:

```
> memfs-a64.exe -m C:\MOUNT
cannot mount MEMFS
The service memfs has failed to start (Status=c000000d).
```

However, `memfs-a64.exe -m M:` works just fine.

Is this a known issue on ARM? Can it be solved or is it a limitation we have to live with?

**Comment by billziss-gh** [comment]
This is not a known issue. Mounting on directories should work the same as on x64 systems.

- Does the `C:\MOUNT` directory already exist? (It should not.)
- Does the file system process creating the `C:\MOUNT` directory have sufficient privileges to do so?

Unfortunately I do not currently have a way for running the full test suite on ARM64 in an automated fashion and the ARM64 OS **does** have differences from the x64 one (e.g. see [this](https://github.com/winfsp/winfsp/blob/v1.11/src/sys/devtimer.c#L24-L28)). So it is possible that I have missed this.

**Reply by overheadhunter** [reply]
> * Does the `C:\MOUNT` directory already exist? (It should not.)

It does not (we did try both, though).

> * Does the file system process creating the `C:\MOUNT` directory have sufficient privileges to do so?

Well in fact we used `C:\Users\...\Desktop\MOUNT` with the user being the same that ran the process.

**Reply by billziss-gh** [reply]
This should work. If it does not it is a bug.

I will create an issue from this discussion.
